### PR TITLE
Swerve velocity control testing

### PR DESCRIPTION
**Pull request by Tan560**
Just created a branch to test velocity control for swerve.  Code Gen will need to account for both sets of gains. If code gen doesn't get it done then just use XML values

@witcpalek  Please comment on any other issues you see with implementation

### CODE REVIEW

File: src/main/cpp/chassis/SwerveModule.cpp [hunk 1]
```diff
@@ -183,13 +184,21 @@ void SwerveModule::RunCurrentState()
...
-    // Run Open Loop
-    m_activeState.speed = (abs(speed.to<double>() / m_maxSpeed.to<double>()) < 0.05) ? 0_mps : speed;
-    // convert mps to unitless rps by taking the speed and dividing by the circumference of the wheel
-    auto driveTarget = m_activeState.speed.to<double>() / (units::length::meter_t(m_wheelDiameter).to<double>() * numbers::pi);
-    auto percent = driveTarget / m_maxSpeed.to<double>();
-    DutyCycleOut out{percent};
-    m_driveTalon->SetControl(out);
+    if (m_velocityControlled)
+    {
+        m_activeState.speed = (abs(speed.to<double>() / m_maxSpeed.to<double>()) < 0.05) ? 0_mps : speed;
```
Comment (witcpalek) [inline src/main/cpp/chassis/SwerveModule.cpp:189]: This line is common between both branches of the if/else, so pull it out before the if

File: src/main/cpp/chassis/SwerveModule.cpp [hunk 2]
```diff
@@ -260,6 +269,21 @@ void SwerveModule::InitDriveMotor(bool driveInverted)
     {
         m_driveTalon->GetConfigurator().Apply(TalonFXConfiguration{}); // Apply Factory Defaults
 
+        configs::TalonFXConfiguration configs{};
+
+        configs.Voltage.PeakForwardVoltage = 11;
+        configs.Voltage.PeakReverseVoltage = -11;
+
+        /* Torque-based velocity does not require a feed forward, as torque will accelerate the rotor up to the desired velocity by itself */
+        /// TO DO : Need code gen updates to be able to be implemented
```
Comment (witcpalek) [inline src/main/cpp/chassis/SwerveModule.cpp:278]: You can add to the XML file for now and then code gen knows what to create

File: src/main/cpp/chassis/SwerveModule.cpp [hunk 3]
```diff
@@ -260,6 +269,21 @@ void SwerveModule::InitDriveMotor(bool driveInverted)
...
+        configs.Slot1.kS = 2.5;       // To account for friction, add 2.5 A of static feedforward
+        configs.Slot1.kP = m_driveKp; // An error of 1 rotation per second results in 5 A output
+        configs.Slot1.kI = m_driveKi; // No output for integrated error
+        configs.Slot1.kD = m_driveKd; // No output for error derivative
+        // Peak output of 40 A
```
Comment (witcpalek) [inline src/main/cpp/chassis/SwerveModule.cpp:283]: Should some of these settings be dependent upon the control mode?

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/cpp/chassis/SwerveModule.cpp (hunk 3 above)
Comment (Tan560) [reply src/main/cpp/chassis/SwerveModule.cpp:283]: Sorry what do you mean by that? LIke if it is Velocity Voltage controlled or if it should be Torque controlled?

File: src/main/cpp/chassis/SwerveModule.cpp (hunk 1 above)
Comment (Tan560) [reply src/main/cpp/chassis/SwerveModule.cpp]: I have made the fix

**Pull request closed without merging**